engine/tests/phpunit/Elgg/I18n/TranslatorTest.php

```diff
@@ -12,7 +12,34 @@ public function testSetLanguageFromGetParameter() {
 		_elgg_services()->input->set('hl', $input_lang);
 		
 		$lang = $translator->getLanguage();
-		$this->assertEquals($lang, $input_lang);		
+		$this->assertEquals($lang, $input_lang);
 	}
 	
+	public function testKeyIsReturnedIfNoTranslationCanBeFound() {
+		$translator = new Translator();
+		
+		$this->assertEquals('foobar', $translator->translate('foobar'));
+	}
+
+	public function testTranslateReturnsTranslationForSpecifiedLanguageIfAvailable() {
+		$this->markTestIncomplete();
+	}
+	
+	public function testTranslateReturnsTranslationForUserLanguageIfNoLanguageWasSpecified() {
+		$this->markTestIncomplete();
+	}
+
+	public function testFallsBackToUserLanguageIfTranslationForSpecifiedLanguageIsNotAvailable() {
+		$this->markTestIncomplete();
+	}
+	
+	public function testDoesNotPerformSprintfFormattingIfArgsNotProvided() {
```

Comment:
This test should give some idea of how I envision this working.

Eventually we will move most of the logic for loading the translation resources into a MessageBundle class.